src/instances.ts

```diff
@@ -87,23 +87,35 @@ function createFilePathKeyMapper(
   compiler: typeof typescript,
   loaderOptions: LoaderOptions
 ) {
+  // Cache file path key - a map lookup is much faster than filesystem/regex operations & the result will never change
+  const filePathMapperCache = new Map<string, FilePathKey>();
   // FileName lowercasing copied from typescript
   const fileNameLowerCaseRegExp = /[^\u0130\u0131\u00DFa-z0-9\\/:\-_\. ]+/g;
   return useCaseSensitiveFileNames(compiler, loaderOptions)
     ? pathResolve
     : toFileNameLowerCase;
 
   function pathResolve(x: string) {
-    return path.resolve(x) as FilePathKey;
+    let cachedPath = filePathMapperCache.get(x);
+    if (!cachedPath) {
+      cachedPath = path.resolve(x) as FilePathKey;
+      filePathMapperCache.set(x, cachedPath);
```

Comment:
Could we rename `x` to something more meaningful please?  I realise you didn't create that name, but reviewing the PR has made realise how it's not as helpful as could be.

Reply:
Will do. I wasn't sure how scoped you'd like changes to be in individual PRs

Reply:
Are the inputs relative paths? I’m surprised this has such an impact. Nice investigating 👍

Reply:
That's a great question - can you confirm @berickson1?

Reply:
They are absolute paths. But path.resolve still does string normalization on every call (which results in one string allocation per character in the path and at least one string concat - and string allocation - operation per folder in the path)